src/pudl/metadata/classes.py

```diff
@@ -1851,6 +1862,14 @@ def from_data_source_ids(
             for res in pkg.resources
             if res.etl_group in data_source_ids + extra_etl_groups
         ]
+
+        # Get XBRL based resources
+        for xbrl_id in xbrl_ids:
+            with open(pudl_settings[f"{xbrl_id}_descriptor"]) as f:
+                descriptor = json.load(f)
+
+            resources.extend(Package(**descriptor).resources)
```

Comment:
Took a lot of work to get to this one-liner!

Reply:
I split it up and added comments